### DISCUSSION TITLE
Why GPy and not e.g. sklearn's GP or GPFlow?

### DISCUSSION THREAD

**Discussion by ekalosak** [SheffieldML/GPy#962] [General]
It's a little bit of an elephant in the room if we don't address it head on: what about GPy makes it better, faster, smarter, or x than its alternatives? I'm an avid GPy fan and user, professionally and personally, but I find myself asking this question and it _is_ somewhat pressing as the GPy project ages.

Some considerations are:
- GPFlow requires tensorflow - that's bulky.. But it's warranted if it makes kernel implementation easier with auto-differentiation - for example.
- GPy works natively with GPyOpt, but the latter is deprecated.
- Paramz is an achievement, but it's convoluted and makes the deeper parts of GPy hard to use (e.g. explicit priors).

I _know_ I'm not covering nearly enough of GPy's functionality to have a fully fleshed discussion, so I'd love to learn more from other users about comparisons and pain-points they've seen.

Ultimately, if this discussion is resolved to a set of key advantages, we need to put them up front and center (main README.md?) in a comparisons table, the latter being commonplace in confident projects with available alternatives.

**Comment by igi-chris** [comment]
For me, one of the key advantages of GPy over the SKLearn implementation is the SparseGP options. With that being said I did want to continue using the wider SKLearn ecosystem with Pipelines, preprocessing etc, so I am building an SKLearn wrapper. I would be happy to contribute this if it is useful to anyone.
